tooling/theme-typings/src/extract-color-schemes.ts

```diff
@@ -0,0 +1,37 @@
+import { isObject } from "@chakra-ui/utils"
+
+const colorHueKeys = [
+  "50",
+  "100",
+  "200",
+  "300",
+  "400",
+  "500",
+  "600",
+  "700",
+  "800",
+  "900",
+]
```

Comment:
What happens when a user defines custom color keys like

```js
const colors = {
   light: "",
   dark: ""
}
```

Is this a scenario we need to account for?

Reply:
Yes, the interface generator accounts for this scenario. Please see this test case: 

https://github.com/chakra-ui/chakra-ui/blob/3591e33f522c1515e71562d45f66bc19cfc6ae43/tooling/theme-typings/test/theme-typings.test.ts#L9-L11

https://github.com/chakra-ui/chakra-ui/blob/3591e33f522c1515e71562d45f66bc19cfc6ae43/tooling/theme-typings/test/theme-typings.test.ts#L75